Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.cpp

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntryUtils.h>
+
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/string/string_view.h>
+#include <AzCore/Serialization/Utils.h>
+#include <AzCore/IO/GenericStreams.h>
+#include <AzCore/IO/FileIO.h>
+
+#include <QMimeData>
+#include <QUrl>
+#include <AzToolsFramework/AssetBrowser/Entries/AssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/SourceAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/ProductAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/FolderAssetBrowserEntry.h>
+#include <AzToolsFramework/AssetBrowser/Entries/RootAssetBrowserEntry.h>
+
+namespace AzToolsFramework
+{
+    namespace AssetBrowser
+    {
+        namespace Utils
+        {
+            void ToMimeData(QMimeData* mimeData, const AZStd::vector<const AssetBrowserEntry*>& entries)
+            {
+                if (!mimeData)
```

Comment:
I could, yeah.   This was only done to be compatible with the previous API.

Reply:
Actually I started doing this and had to add too many more checks all over the place, because it turns out the originator of these calls is Qt itself, and the Qt layer initiates all these calls to various locations by calling callbacks with a QMimeData* or const QMimeData*.  we can't change Qt's API.

  This means that it ends up having to check every place instead of just here in 2 places... so I think I'll leave it.